### PR TITLE
iserver-test: Fix BC 9.0 test

### PR DESCRIPTION
**Pull request by ericywl**
## Motivation/summary

8.19 cannot upgrade to 9.0, so our BC test for 9.0 will always fail. This PR adds an edge case handling for 9.0, and also increase the APM server ready wait time.

## How to test these changes

Run workflow: https://github.com/elastic/apm-server/actions/runs/16211738588

### CODE REVIEW

File: .github/workflows/generate-bc-upgrade-paths/bc.sh [hunk 1]
```diff
@@ -15,6 +15,12 @@ for bc in ${bcs}; do
     if [[ ${major} -lt 8 ]]; then
         continue
     fi
+    # Specifically for 9.0, we set the upgrade path to be from 8.18,
+    # since 8.19 cannot upgrade to 9.0.
```
Comment (marclop) [inline .github/workflows/generate-bc-upgrade-paths/bc.sh]: Why's that? Is there a legitimate reason for that?

Comment (ericywl) [reply .github/workflows/generate-bc-upgrade-paths/bc.sh]: Its due to ES upgrade policy, upgrades can only be done in chronological order. Since latest 9.0 was released before latest 8.19, it cannot be upgraded to.

ES issue: https://elasticco.atlassian.net/browse/CP-10254.

Comment (marclop) [reply .github/workflows/generate-bc-upgrade-paths/bc.sh]: But 8.19 can be upgraded to 9.1?

Comment (ericywl) [reply .github/workflows/generate-bc-upgrade-paths/bc.sh]: Yes.

Comment (marclop) [reply .github/workflows/generate-bc-upgrade-paths/bc.sh]: Can you add this information in the comment, please? I'm sure others may have the same questions.
